### PR TITLE
feat(accesstoken): Add project access token resource

### LINKED ISSUE #58: Support  gitlab project access tokens resource.
### What problem are you facing?

In our environment we use some automation flow to create a gitops projects in a local gitlab instance,  it’d be great to have an opportunity also create  [gitlab project access token](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html)  for each gitops project to use it then in our main CI/CD pipeline.

### How could Crossplane help solve your problem?
To solve this it requires  to implement one more controller in GitLab Provider  to manage access token resource in Crossplane way. 
If it sounds reasonable  I'd  like to implement this functionality.

### PR DESCRIPTION
**Pull request by IntaSlapna**
### Description of your changes

Fixes #58

I have:

- [ ] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by MisterMX — CHANGES_REQUESTED** [review]

File: pkg/controller/projects/accesstokens/controller.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+	at, _, err := e.client.GetProjectAccessToken(*cr.Spec.ForProvider.ProjectID, AccessTokenID)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(isErrorProjectAccessTokenNotFound, err), errGetFailed)
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	lateInitializeProjectAccessToken(&cr.Spec.ForProvider, at)
+
+	cr.Status.AtProvider = v1alpha1.AccessTokenObservation{}
```
[the remark this answers is not in the document]
Comment (IntaSlapna) [reply pkg/controller/projects/accesstokens/controller.go:119]: deleted

File: pkg/controller/projects/accesstokens/controller.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+	}
+
+	if in.ExpiresAt == nil && accessToken.ExpiresAt != nil {
+		in.ExpiresAt = &metav1.Time{Time: time.Time(*accessToken.ExpiresAt)}
+	}
+}
+
+// findAccessToken try to find a access token with the ID in the access token array,
+// if found return a access token otherwise return nil.
+func findAccessToken(accessTokenID int, accessTokens []*gitlab.ProjectAccessToken) *gitlab.ProjectAccessToken {
```
[the remark this answers is not in the document]
Comment (IntaSlapna) [reply pkg/controller/projects/accesstokens/controller.go:206]: deleted

File: pkg/controller/projects/accesstokens/controller.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+// IsErrorProjectAccessTokenNotFound helper function to test for errProjectAccessTokenNotFound error.
+func isErrorProjectAccessTokenNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), errProjecAccessTokentNotFound)
+}
+
+// NewAccessTokenClient returns a new Gitlab ProjectAccessToken service
+func newAccessTokenClient(cfg clients.Config) gitlab.ProjectAccessTokensService {
```
[the remark this answers is not in the document]
Comment (IntaSlapna) [reply pkg/controller/projects/accesstokens/controller.go:226]: deleted

File: pkg/controller/projects/accesstokens/controller.go [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+			return v
+		}
+	}
+	return nil
+}
+
+//-- What was from interface file --//
+
+// IsErrorProjectAccessTokenNotFound helper function to test for errProjectAccessTokenNotFound error.
+func isErrorProjectAccessTokenNotFound(err error) bool {
```
[the remark this answers is not in the document]
Comment (IntaSlapna) [reply pkg/controller/projects/accesstokens/controller.go:218]: Deleted from clients/projects/accesstoken.go, left in controller.go

[the remark this answers is not in the document]
Comment (IntaSlapna) [reply pkg/controller/projects/accesstokens/controller.go:218]: Deleted and imported from clients/projects.

**Review by MisterMX — CHANGES_REQUESTED** [review]

**Review by MisterMX — APPROVED** [review]
LGTM. Thank you very much for your contribution @IntaSlapna!

**Pull request merged**